rollingpin/commands.py

```diff
@@ -0,0 +1,82 @@
+from abc import ABCMeta, abstractproperty
+
+
+class Command(object):
+    __metaclass__ = ABCMeta
+
+    CONTINUE = 1
+    SKIP_REMAINING = 2
+
+    def __init__(self, args=None):
+        if args is None:
+            self._args = []
+        else:
+            self._args = args
+
+    @abstractproperty
+    def name(self):
+        raise NotImplementedError
+
+    @property
+    def args(self):
+        return self._args
+
+    def add_argument(self, arg):
+        self._args.append(arg)
+
+    def cmdline(self):
+        return [self.name()] + self._args
+
+    def check_result(self, result):
+        return Command.CONTINUE
+
+    def __str__(self):
+        return "Command(name={}, args={})".format(self.name(), self._args)
```

Comment:
probably wanna leave this under `__repr__` only if this class doesn't have a natural string representation by itself (maybe the command line? but that's a bit weird)